### ISSUE TITLE
Formant Filter - center frequency following note pitch?

### ISSUE THREAD

**Issue by unfa** [zynaddsubfx/zyn-fusion-issues#102]
When messing with the Formant filter I've got an idea - would it be possible to add a control to make the frequency offset of the whole formant follow the note pitch?

So essentially playing a C4 will place the formant resonances in spot f but playing a C5 note will place them in f*2.

This could be used to create sounds resembling "sampled" synths - where the filtering and EQ moves with the pitch of the note, because the whole sample is played back at a different rate.

A quick mockup:

![selection_893](https://user-images.githubusercontent.com/491704/28407319-feec1ebc-6d33-11e7-9004-6ee97728fdfc.png)

**Comment by fundamental** [comment]
Perhaps I'm missing something, but isn't there already a frequency tracking knob in place which is shared with other filter types? In the fltk UI it still uses the lower resolution version of this parameter, but in the fusion UI it goes from -100% to +100% frequency tracking which should produce the behavior that you're looking for, no?

**Comment by unfa** [comment]
Yes, there is a Frequency tracking feature available for all filters, but for the Formant Filter it'll only affect the currently used vovel (the BS Position) - and what I mean here is changing the Center Frequency of the whole thing up and down with note pitch, without changing the BS Position.

I guess it then should also respond to portamento and such. I wonder if it should follow oscillator pitch envelope and LFO modulation - that should probably be controlled separately if so. One might want a sound that has vibrato on the osciallators, without making the formant vibrate as well. But someone might want to use a pitch envelope and make the formant center frequency follow this.

Now I wonder if this is even relevant without also affecting the Octave Span (I made up that name) to produce correctresults...
